packages/front-end/services/track.ts

```diff
@@ -109,7 +109,9 @@ function getOrGenerateDeviceId() {
 }
 
 function getOrGeneratePageId() {
-  if (!(window.history.state.key in pageIds)) {
+  // On initial load if the router hasn't initialized a state change yet then history.state will be null.
+  // Since this only happens on one pageload, using a hardcoded default key should still work as its own key
+  if (!((window.history.state?.key || "null") in pageIds)) {
```

Comment:
could we just use an empty string here?
```suggestion
  if (!((window.history.state?.key || "") in pageIds)) {
```